### DISCUSSION TITLE
Some Python depencendies (pep621, pip_requirements at least) are not listed and treated anymore when there is no current version set

### DISCUSSION THREAD

**Discussion by Lowaiz** [renovatebot/renovate#23945] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.51.3

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

TL DR: After watching this [old discussion](https://github.com/renovatebot/renovate/discussions/22920), it seems normal that renovate skips the unpinned deps, but in my memory, Renovate was trying to pin to the latest available version when that happen, and I can't find a way to get this behavior back. 

Dependencies inside Python's managers files are totally skipped, even from the DependencyDashboard.
So Renovate doesn't propose to pin the versions, even with the `rangeStrategy` set to `pin`.
I tried to add a rangeStrategy to the configuration, to the python manager configuration, the pep621,... and tried few other options to make Renovate treats these deps, but it seems useless.

reproduction: https://github.com/Lowaiz/renovate-bug

### Relevant debug logs

<details><summary>Logs</summary>

```
"pep621": [
           {
             "extractedConstraints": {"python": ">=3.11"},
             "deps": [
               {
                 "packageName": "requests",
                 "depName": "requests",
                 "datasource": "pypi",
                 "depType": "project.dependencies",
                 "skipReason": "unspecified-version",
                 "updates": []
               },
               {
                 "packageName": "gql",
                 "depName": "gql",
                 "datasource": "pypi",
                 "depType": "project.dependencies",
                 "skipReason": "unspecified-version",
                 "updates": []
               },
               {
                 "packageName": "pyyaml",
                 "depName": "pyyaml",
                 "datasource": "pypi",
                 "depType": "project.dependencies",
                 "skipReason": "unspecified-version",
                 "updates": []
               }
             ]  
"pip_requirements": [
           {
             "deps": [
               {
                 "depName": "requests",
                 "datasource": "pypi",
                 "updates": [],
                 "packageName": "requests",
                 "versioning": "pep440",
                 "warnings": [],
                 "skipReason": "invalid-value"
               },
               {
                 "depName": "pyyaml",
                 "datasource": "pypi",
                 "updates": [],
                 "packageName": "pyyaml",
                 "versioning": "pep440",
                 "warnings": [],
                 "skipReason": "invalid-value"
               }
             ],
             "packageFile": "requirements.txt"
           }
         ]
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
I don't recall renovate ever supporting pinning in this scenario
